resteasy-jaxrs/src/main/java/org/jboss/resteasy/microprofile/config/ResteasyConfig.java

```diff
@@ -102,20 +137,38 @@ public String getValue0(String propertyName, SOURCE source, String defaultValue)
       }
       else
       {
-         return config.getOptionalValue(propertyName, String.class).orElse(defaultValue);
+         try
+         {
+            final Object config = GET_CONFIG.invoke(null, getClassLoader());
+            Optional<String> opt = (Optional<String>) GET_OPTIONAL_VALUE.invoke(config, propertyName, String.class);
+            return opt.orElse(defaultValue);
+         }
+         catch (IllegalAccessException | InvocationTargetException e) {
+            LogMessages.LOGGER.debugf(e, "Failed to invoke the configuration API method %s.", GET_VALUE);
+            return defaultValue;
+        }
       }
    }
 
    public Iterable<String> getPropertyNames()
    {
-      return config == null ? null : getPropertyNames0();
+      return !MP_CONFIG_PRESENT ? null : getPropertyNames0();
    }
 
+   @SuppressWarnings("unchecked")
    public Iterable<String> getPropertyNames0()
    {
       if (System.getSecurityManager() == null)
```

Comment:
While we already do this, I'm not sure we should be wrapping this in a privileged action as this is publicly accessible.

Reply:
Hmmm. But if I take that out, the method won't work if there's a SecurityManager ... .

Reply:
Well it would delegate the work to the config API. By doing it here we are allowing unchecked access to the property names. Currently for SmallRye this isn't an issue, but if they ever decide to add a check it could be an issue.

Reply:
> it would delegate the work to the config API

I'm not following. Isn't the call to GET_PROPERTY_NAMES wrapped in a PrivilegedExceptionAction because it's a reflective call?

Reply:
`Method.invoke()` does not require special permissions so we'd override any permission checking in the `Confg.getPropertyNames()`.